src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/matchtoterm/TermFieldRewriter.java

```diff
@@ -147,19 +149,27 @@ public boolean visit(SQLIdentifierExpr expr) {
 
     public void collect(SQLTableSource tableSource, Map<String, String> indexToType,  Map<String, String> aliases) {
         if (tableSource instanceof SQLExprTableSource) {
+
             String table = null;
-            if (((SQLExprTableSource) tableSource).getExpr() instanceof SQLIdentifierExpr) {
-                table = ((SQLIdentifierExpr) ((SQLExprTableSource) tableSource).getExpr()).getName();
-                indexToType.put(
-                    table,
-                    null
-                );
-            } else if (((SQLExprTableSource) tableSource).getExpr() instanceof SQLBinaryOpExpr) {
-                table = ((SQLIdentifierExpr)((SQLBinaryOpExpr) ((SQLExprTableSource) tableSource).getExpr()).getLeft()).getName();
-                indexToType.put(
-                    table,
-                    ((SQLIdentifierExpr)((SQLBinaryOpExpr) ((SQLExprTableSource) tableSource).getExpr()).getRight()).getName()
-                );
+            SQLExprTableSource sqlExprTableSource = (SQLExprTableSource) tableSource;
+
+            if (sqlExprTableSource.getExpr() instanceof SQLIdentifierExpr) {
+                SQLIdentifierExpr sqlIdentifier = (SQLIdentifierExpr) sqlExprTableSource.getExpr();
+                table = sqlIdentifier.getName();
+                indexToType.put(table, null);
+            } else if (sqlExprTableSource.getExpr() instanceof SQLBinaryOpExpr) {
+                SQLBinaryOpExpr sqlBinaryOpExpr = (SQLBinaryOpExpr) sqlExprTableSource.getExpr();
+                table = ((SQLIdentifierExpr) sqlBinaryOpExpr.getLeft()).getName();
+                SQLExpr rightSideOfExpression = sqlBinaryOpExpr.getRight();
```

Comment:
np: `rightTerm`

Reply:
As we can see, on the right side there can be lots of different things (SQLIdentifier, SQLCharExpression), so I'd like to keep it as is. If you insist, though, I'll rename

Reply:
no objections, that's why it's `np`